morphir/ir/src/morphir/ir/distribution/Codec.scala

```diff
@@ -0,0 +1,82 @@
+package morphir.ir.distribution
+
+import morphir.FormatVersion
+import morphir.ir.Distribution.Distribution
+
+/**
+ * Generated based on IR.Distribution
+ */
+object Codec {
+
+  val encodeDistribution: io.circe.Encoder[morphir.ir.Distribution.Distribution] = (
+    (distribution: morphir.ir.Distribution.Distribution) =>
+      distribution match {
+        case morphir.ir.Distribution.Library(arg1, arg2, arg3) =>
+          io.circe.Json.arr(
+            io.circe.Json.fromString("Library"),
+            morphir.ir._package.Codec.encodePackageName(arg1),
+            morphir.sdk.dict.Codec.encodeDict(
+              morphir.ir._package.Codec.encodePackageName,
+              morphir.ir._package.Codec.encodeSpecification(morphir.sdk.basics.Codec.encodeUnit)
+            )(arg2),
+            morphir.ir._package.Codec.encodeDefinition(
+              morphir.sdk.basics.Codec.encodeUnit,
+              morphir.ir._type.Codec.encodeType(morphir.sdk.basics.Codec.encodeUnit)
+            )(arg3)
+          )
+      }
+    )
+
+  implicit val encodeVersionDistribution: io.circe.Encoder[morphir.ir.Distribution.Distribution] =
+    (distribution: morphir.ir.Distribution.Distribution) =>
+      io.circe.Json.obj(("formatVersion", io.circe.Json.fromInt(FormatVersion.formatVersion)),
+        ("distribution", encodeDistribution(distribution)))
+
+
+  implicit val decodeVersionDistribution: io.circe.Decoder[Distribution] =
+    (c: io.circe.HCursor) =>
+      c.downField("formatVersion").as[Int]
+        .flatMap{
+          case FormatVersion.formatVersion => c.downField("distribution").as(decodeDistribution)
+          case _ => Left(io.circe.DecodingFailure("IR version is old, please re generate IR",Nil))
+        }
```

Comment:
Can we also move this to a separate file to avoid overwriting during code generation?